File: data/StarCoder2/readable_format_HPEC/fortran/file279.f90
program DRB014_outofbounds_orig_yes
    use omp_lib
    implicit none

    integer :: i, j, n, m
    real, dimension (:,:), allocatable :: b

    n = 100
    m = 100

    allocate (b(n,m))

    !$omp parallel do private(i)
    do j = 2, n
        do i = 1, m
            b(i,j) = b(i-1,j)
        end do
    end do
    !$omp end parallel do
    print*,"b(50,50)=",b(50,50)

    deallocate(b)
end program

File: data/llama3_8B_ins_results/readable_format_HPEC/cpp/file279.cpp
#include <iostream>
#include <vector>
#include <omp.h>

int main() {
    int n = 100;
    int m = 100;

    std::vector<std::vector<double>> b(n, std::vector<double>(m));

    #pragma omp parallel for private(int i)
    for (int j = 2; j < n; j++) {
        for (int i = 1; i <= m; i++) {
            b[i-1][j] = b[i-2][j];
        }
    }

    std::cout << "b(50,50) = " << b[50][50] << std::endl;

    return 0;
}
